ToolBox/LeanDataReader.cs

```diff
@@ -48,6 +52,66 @@ public LeanDataReader(SubscriptionDataConfig config, Symbol symbol, Resolution r
             _config = config;
         }
 
+        /// <summary>
+        /// Initialize a instance of LeanDataReader from a path to a zipped data file.
+        /// It also supports declaring the zip entry CSV file for options and futures.  
+        /// </summary>
+        /// <param name="filepath">Absolute or relative path to a zipped data file, optionally the zip entry file can be declared by using '#' as separator.</param>
+        /// <example>
+        /// var dataReader = LeanDataReader("../relative/path/to/file.zip")
+        /// var dataReader = LeanDataReader("absolute/path/to/file.zip#zipEntry.csv")
+        /// </example>
+        public LeanDataReader(string filepath)
+        {
+            Symbol symbol;
+            DateTime date;
+            Resolution resolution;
+            string zipEntry = null;
+
+            var localDiskMapFileProvider = new LocalDiskMapFileProvider();
+            Composer.Instance.AddPart<IMapFileProvider>(localDiskMapFileProvider);
+
+            var isFutureOrOption = filepath.Contains("#");
+
+            if (isFutureOrOption)
+            {
+                zipEntry = filepath.Split('#')[1];
+                filepath = filepath.Split('#')[0];
+            }
+            
+            var fileInfo = new FileInfo(filepath);
+            if (!LeanData.TryParsePath(fileInfo.FullName, out symbol, out date, out resolution))
+            {
+                throw new ArgumentException($"File {filepath} cannot be parsed.");
+            }
+
+            if (isFutureOrOption)
+            {
+                symbol = LeanData.ReadSymbolFromZipEntry(symbol, resolution, zipEntry);
+            }
+
+            var marketHoursDataBase = MarketHoursDatabase.FromDataFolder();
+            var dataTimeZone = marketHoursDataBase.GetDataTimeZone(symbol.ID.Market, symbol, symbol.SecurityType);
+            var exchangeTimeZone = marketHoursDataBase.GetExchangeHours(symbol.ID.Market, symbol, symbol.SecurityType).TimeZone;
+
+            var tickType = LeanData.GetCommonTickType(symbol.SecurityType);
+            var fileName = Path.GetFileNameWithoutExtension(fileInfo.Name);
+            if (fileName.Contains("_"))
+            {
+                tickType = (TickType)Enum.Parse(typeof(TickType), fileName.Split('_')[1], true);
```

Comment:
should tick type resolution be added to `TryParsePath` ? -- we don't need to make this change now, but it's food for thought. Reasonably the tick type is part of the path, so we could add another `out` parameter w/ the tick type, defaulting to the common tick type if not present in the file name.